src/node_util.cc

```diff
@@ -248,6 +248,13 @@ static void GuessHandleType(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(OneByteString(env->isolate(), type));
 }
 
+
+// v8::Private is actually just a Symbol with some special behaviour.
```

Comment:
I’m not a fan of exposing this, and I’m pretty sure neither is the V8 team (they may `v8::Private` not inherit from `v8::Value` for a reason).

Reply:
@nodejs/v8 if there's a better way to do a createPrivateSymbol without using natives i'm up for it. I just want to avoid weakmaps.

Reply:
I was talking about using private symbols from JS land in general, the method of creating them here would be just fine imo.

Reply:
well they already expose such a function to natives (createPrivateSymbol, it's used in chromium for readable streams) but it isn't accessable from our c++ or js.

Reply:
Is there a compelling reason to use the v8::Private here?

Also process.binding('util') is public BTW.

Reply:
normal symbols aren't actually private, and ModuleWrap should be inaccessible from userland

Reply:
I would still think that an unenumerable JS symbols should be sufficient here, we do that a lot in other places and it makes accessing the symbol reasonably hard imo

Reply:
I'd rather it wasn't exposed at all... it's an internalBinding API for a reason.

Reply:
I guess the question is, do we have evidence that using normal symbols is not enough for our usual requirement of privacy, and people do exploit them (e.g. through `Reflect.ownKeys`) in the wild?

Reply:
I mean if we agree here that this leaking won't make future changes semver major even if it breaks stuff that's fine with me too.

I'd imagine people are a bit more wary about this now given the whole grub incident, but I wouldn't underestimate people using a module intented for low level manipulation of javascript.

Reply:
What made you change your mind about private class fields?

Reply:
@targos sharing data between superclass and subclass can't be done with private fields (at least not in any simple way worth maintaining)